contracts/cw-proposal-multiple/src/contract.rs

```diff
@@ -0,0 +1,672 @@
+        info.sender.to_string(),
+        vote.to_string(),
+    )?;
+    Ok(Response::default()
+        .add_submessages(change_hooks)
+        .add_submessages(vote_hooks)
+        .add_attribute("action", "vote")
+        .add_attribute("sender", info.sender)
+        .add_attribute("proposal_id", proposal_id.to_string())
+        .add_attribute("position", vote.to_string())
+        .add_attribute("status", prop.status.to_string()))
+}
+
+pub fn execute_execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    proposal_id: u64,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    if config.only_members_execute {
+        let power = get_voting_power(
+            deps.as_ref(),
+            info.sender.clone(),
+            config.dao.clone(),
+            Some(env.block.height),
+        )?;
+        if power.is_zero() {
+            return Err(ContractError::Unauthorized {});
+        }
+    }
+
+    let mut prop = PROPOSALS
+        .may_load(deps.storage, proposal_id)?
+        .ok_or(ContractError::NoSuchProposal { id: proposal_id })?;
+
+    // Check here that the proposal is passed. Allow it to be
+    // executed even if it is expired so long as it passed during its
+    // voting period.
+    let old_status = prop.status;
+    if !prop.is_passed() {
+        return Err(ContractError::NotPassed {});
+    }
+    prop.status = Status::Executed;
+    PROPOSALS.save(deps.storage, proposal_id, &prop)?;
+
+    let return_deposit = get_return_deposit_msg(&prop)?;
+    let (_, winning_choice) = prop.calculate_winning_choice();
+
+    let response = if !winning_choice.msgs.is_empty() {
+        let execute_message = WasmMsg::Execute {
+            contract_addr: config.dao.to_string(),
+            msg: to_binary(&cw_core::msg::ExecuteMsg::ExecuteProposalHook {
+                msgs: winning_choice.msgs.clone(),
+            })?,
+            funds: vec![],
+        };
+        Response::<Empty>::default().add_message(execute_message)
+    } else {
+        Response::default()
+    };
+
+    let hooks = proposal_status_changed_hooks(
+        PROPOSAL_HOOKS,
+        deps.storage,
+        proposal_id,
+        old_status.to_string(),
+        prop.status.to_string(),
+    )?;
+    Ok(response
+        .add_messages(return_deposit)
+        .add_submessages(hooks)
+        .add_attribute("action", "execute")
+        .add_attribute("sender", info.sender)
+        .add_attribute("proposal_id", proposal_id.to_string())
+        .add_attribute("dao", config.dao))
+}
+
+pub fn execute_close(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    proposal_id: u64,
+) -> Result<Response<Empty>, ContractError> {
+    let mut prop = PROPOSALS.load(deps.storage, proposal_id)?;
+
+    // Only open and expired or rejected proposals may be closed.
+    match prop.status {
+        Status::Rejected => (),
+        Status::Open => {
+            if !prop.expiration.is_expired(&env.block) {
```

Comment:
```suggestion
            if !prop.expiration.is_expired(&env.block) || prop.is_passed(&env.block) {
```

Can't close a passed proposal.